src/services/review/ReviewRequestService.ts

```diff
@@ -71,70 +100,163 @@ export default class ReviewRequestService {
     this.reviewMeta = reviewMeta
     this.reviewRequestView = reviewRequestView
     this.pageService = pageService
+    this.configService = configService
   }
 
-  compareDiff = async (
-    sessionData: UserWithSiteSessionData,
+  compareDiff = (
+    userWithSiteSessionData: UserWithSiteSessionData,
     stagingLink: StagingPermalink
-  ): Promise<EditedItemDto[]> => {
+  ) =>
+    ResultAsync.fromPromise(
+      this.apiService.getCommitDiff(userWithSiteSessionData.siteName),
+      // TODO: Write a handler for github errors to our own internal errors
+      () => new NetworkError()
+    )
+      .andThen(({ files, commits }) =>
+        ResultAsync.fromPromise(
+          this.computeShaMappings(commits),
+          () => new DatabaseError()
+        ).map((mappings) => ({ mappings, files }))
+      )
+      .andThen(({ mappings, files }) =>
+        combine(
+          this.compareDiffWithMappings(
+            userWithSiteSessionData,
+            stagingLink,
+            files,
+            mappings
+          ).map((res) =>
+            res
+              // NOTE: Need to type-hint so that we can recover
+              .map<BaseEditedItemDto | WithEditMeta<EditedItemDto>>(_.identity)
+              .orElse((baseItem) => okAsync(baseItem))
+          )
+        )
+      )
+
+  private compareDiffWithMappings = (
+    sessionData: UserWithSiteSessionData,
+    stagingLink: StagingPermalink,
+    files: RawFileChangeInfo[],
+    mappings: Record<
+      string,
+      {
```

Comment:
might want to abstract this as a type and use with a readable name

Reply:
caught me being lazy AHHAHHA wdyt about `shaMappings`

Reply:
Hahaha! 😂 

Shall we define one `shaMapping` and then `shaMappings` is a Record of string to `shaMapping`?

Reply:
`Record` takes 2 type arguments, so i think sha mapping would be misleading (cos that's the actual user info that we'd be interested in).

wdyt about 
```ts
type GithubEditInfo = {
        author: string
        unixTime: number
}
type Sha = string
type ShaMappings = Record<Sha, GithubEditInfo>
```

Reply:
okay makes sense to me